src/MahApps.Metro/Controls/NumericUpDown.cs

```diff
@@ -1006,15 +1006,36 @@ private string FormattedValue(double? newValue, string format, CultureInfo cultu
                     if (match.Success)
                     {
                         // we have a format template such as "{0:N0}"
-                        return string.Format(culture, format, ConvertStringFormatValue(newValue.Value, match.Groups["format"].Value));
+                        return string.Format(culture, format, newValue);
                     }
                     // we have a format such as "N0"
-                    var value = ConvertStringFormatValue(newValue.Value, format);
-                    return value.ToString(format, culture);
+                    return newValue.ToString(format, culture);
                 }
             }
 
-            return newValue.Value.ToString(culture);
+            return newValue.ToString(culture);
+        }        
+
+        private double FormattedValue(double newValue, string format, CultureInfo culture)
+        {
+            format = format.Replace("{}", string.Empty);
+            if (!string.IsNullOrWhiteSpace(format))
+            {
+                if (!TryFormatHexadecimal(newValue, format, culture, out string hexValue))
+                {
+                    var match = RegexStringFormat.Match(format);
+                    if (match.Success)
+                    {
+                        // we have a format template such as "{0:N0}"
+                        return ConvertStringFormatValue(newValue, match.Groups["format"].Value);
+                    }
+
+                    // we have a format such as "N0"
+                    return ConvertStringFormatValue(newValue, format);
+                }
+            }
+
+            return newValue;
         }        
 
         private static double ConvertStringFormatValue(double value, string format)
```

Comment:
If we are conforming with the default .NET formatting, do we still need `ConvertStringFormatValue`?

Reply:
Ok, I get it :) We need to handle the conversion the other way now. 100 in text should be 1 as value.